src/pydocstyle/config.py

```diff
@@ -552,8 +653,10 @@ def _get_set(value_str):
             file.
 
             """
+            if not isinstance(value_str, list):
```

Comment:
Well my intention here was so that if `value_str` is for some reason an iterable that is not a `list`, then it should fail instead of silently iterating over it. A concrete example would be something like:
```toml
[tool.pydocstyle.add_ignore]
D100 = true
D200 = false
```
This is an invalid configuration, which would (I think) fail in my version when trying to split the `{'D100': True, 'D200': False}` dictionary value and will silently iterate over the keys in your version.

Although this is a stupid situation in the first place, so I'm not too opposed to changing it to your version.

Reply:
If that's what you're trying to do, I recommend being more explicit in throwing a proper error rather than just leaving nature of fault to deal with it.

```py
if isinstance(value_str, str):
    value_str = value_str.split(",")
elif isinstance(value_str, list):
    pass
else:
    throw ErrorGoesHere("message goes here")
```

Reply:
I moved the check for the case I was worried about to `TomlParser.get` instead.